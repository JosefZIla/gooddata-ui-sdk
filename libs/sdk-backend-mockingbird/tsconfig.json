{
    "compilerOptions": {
        "esModuleInterop": true,
        "target": "es5",
        "module": "commonjs",
        "moduleResolution": "node",
        "lib": ["es6", "dom", "ES2016.Array.Include"],
        "importHelpers": true,
        "declaration": true,
        "declarationMap": true,
        "sourceMap": true,
        "outDir": "./dist",
        "strict": true,
        "noImplicitAny": true,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "noImplicitReturns": true,
        "baseUrl": "..",
        "paths": {
            "@gooddata/sdk-model": ["sdk-model/src"],
            "@gooddata/sdk-backend-base": ["sdk-backend-base/src"],
            "@gooddata/sdk-backend-spi": ["sdk-backend-spi/src"]
        }
    },
    "exclude": ["node_modules", "dist"]
}
