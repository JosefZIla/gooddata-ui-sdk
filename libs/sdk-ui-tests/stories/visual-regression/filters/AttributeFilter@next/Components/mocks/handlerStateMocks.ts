// (C) 2022 GoodData Corporation
import { DefaultState } from "@gooddata/sdk-ui-filters/dist/AttributeFilter@next/hooks/attributeFilterHandlerDefaultResult";
import { IAttributeHandlerState } from "@gooddata/sdk-ui-filters/dist/AttributeFilter@next/hooks/types";

export const stateWithEmptySelection: IAttributeHandlerState = {
    ...DefaultState,
    initialization: {
        status: "success",
    },
    elements: {
        initialPageLoad: {
            status: "success",
        },
        data: [
            { title: "item 1", uri: "uri1" },
            { title: "item 2", uri: "uri2" },
        ],
        nextPageLoad: {
            status: "pending",
        },
        currentOptions: {
            offset: 0,
            limit: 20,
        },
    },
    selection: {
        committed: {
            elements: [],
            keys: [],
            isInverted: false,
        },
        working: {
            elements: [],
            keys: [],
            isInverted: false,
            isChanged: false,
            isEmpty: false,
        },
    },
};

export const stateWithSelectionAndSearch: IAttributeHandlerState = {
    ...DefaultState,
    initialization: {
        status: "success",
    },
    elements: {
        initialPageLoad: {
            status: "success",
        },
        data: [
            { title: "item 1", uri: "uri1" },
            { title: "item 2", uri: "uri2" },
        ],
        nextPageLoad: {
            status: "pending",
        },
        currentOptions: {
            offset: 0,
            limit: 20,
            search: "bla",
        },
    },
    selection: {
        committed: {
            elements: [],
            keys: [],
            isInverted: false,
        },
        working: {
            elements: [{ title: "item 1", uri: "uri1" }],
            keys: ["uri1"],
            isInverted: false,
            isChanged: false,
            isEmpty: false,
        },
    },
};
