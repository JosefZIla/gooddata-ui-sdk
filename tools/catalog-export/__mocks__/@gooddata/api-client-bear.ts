// (C) 2007-2018 GoodData Corporation
import catalogItemsResponse from "./loadItems.json";
import getObjectsResponse from "./getObjects.json";
import getObjectsByQueryResponse from "./getObjectsByQuery.json";
import getVisualizationsResponse from "./getVisualizations.json";
import getAnalyticalDashboards from "./getAnalyticalDashboards.json";

export default {
    catalogue: {
        loadItems() {
            return Promise.resolve(catalogItemsResponse);
        },
    },
    md: {
        getObjectsByQuery() {
            return Promise.resolve(getObjectsByQueryResponse);
        },
        getObjects() {
            return Promise.resolve(getObjectsResponse);
        },
        getVisualizations() {
            return Promise.resolve(getVisualizationsResponse);
        },
        getAnalyticalDashboards() {
            return Promise.resolve(getAnalyticalDashboards);
        },
    },
};
