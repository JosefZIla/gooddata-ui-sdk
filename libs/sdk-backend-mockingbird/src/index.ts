// (C) 2019-2021 GoodData Corporation
/**
 * @packageDocumentation
 */
import { dummyBackend, dummyBackendEmptyData, dummyDataView } from "@gooddata/sdk-backend-base";
export { dummyBackend, dummyBackendEmptyData, dummyDataView };

export { recordedBackend } from "./recordedBackend";

export {
    RecordingIndex,
    ExecutionRecording,
    InsightRecording,
    DisplayFormRecording,
    ScenarioRecording,
    RecordedBackendConfig,
    CatalogRecording,
    VisClassesRecording,
    DashboardRecording,
    RecordedRefType,
    SecuritySettingsUrlValidator,
    SecuritySettingsOrganizationScope,
} from "./recordedBackend/types";

export {
    recordedDataView,
    recordedDataViews,
    recordedInsight,
    DataViewAll,
    dataViewWindow,
    DataViewFirstPage,
    NamedDataView,
} from "./recordedBackend/execution";

export { CompositeBackendPart, compositeBackend } from "./compositeBackend/index";

export {
    LegacyExecutionRecording,
    legacyRecordedBackend,
    legacyRecordedDataView,
    LegacyRecordingIndex,
    LegacyWorkspaceRecordings,
} from "./legacyRecordedBackend";
