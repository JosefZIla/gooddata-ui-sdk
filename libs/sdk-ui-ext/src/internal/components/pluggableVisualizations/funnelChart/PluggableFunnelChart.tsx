// (C) 2019-2022 GoodData Corporation
import React from "react";
import { render } from "react-dom";
import { IVisConstruct, IReferencePoint, IExtendedReferencePoint } from "../../../interfaces/Visualization";

import { PluggablePieChart } from "../pieChart/PluggablePieChart";
import { setFunnelChartUiConfig } from "../../../utils/uiConfigHelpers/funnelChartUiConfigHelper";
import UnsupportedConfigurationPanel from "../../configurationPanels/UnsupportedConfigurationPanel";
import { VisualizationTypes } from "@gooddata/sdk-ui";
import { ISortConfig } from "../../../interfaces/SortConfig";

/**
 * PluggableFunnelChart
 *
 * ## Buckets
 *
 * | Name     | Id       | Accepts             |
 * |----------|----------|---------------------|
 * | Measures | measures | measures only       |
 * | ViewBy   | view     | attribute or date   |
 *
 * ### Bucket axioms
 *
 * - |ViewBy| ≤ 1
 * - |Measures| ≥ 1 ∧ ≤ 20
 * - |ViewBy| = 1 ⇒ |Measures| = 1
 * - |ViewBy| = 0 ⇒ |Measures| ≥ 1
 *
 * ## Dimensions
 *
 * The PluggableFunnelChart always creates two dimensional execution.
 *
 * - |ViewBy| = 0 ⇒ [[], [MeasureGroupIdentifier]]
 * - |ViewBy| = 1 ⇒ [[MeasureGroupIdentifier], [ViewBy]]
 *
 * ## Default sorts
 *
 * The PluggableFunnelChart does not use any sorts.
 *
 */
export class PluggableFunnelChart extends PluggablePieChart {
    constructor(props: IVisConstruct) {
        super(props);
        this.type = VisualizationTypes.FUNNEL;
    }

    public getExtendedReferencePoint(referencePoint: IReferencePoint): Promise<IExtendedReferencePoint> {
        return super.getExtendedReferencePoint(referencePoint).then(setFunnelChartUiConfig);
    }

    protected renderConfigurationPanel(): void {
        if (document.querySelector(this.configPanelElement)) {
            const properties = this.visualizationProperties ?? {};

            render(
                <UnsupportedConfigurationPanel
                    locale={this.locale}
                    pushData={this.pushData}
                    properties={properties}
                />,
                document.querySelector(this.configPanelElement),
            );
        }
    }

    public getSortConfig(_referencePoint: IReferencePoint): Promise<ISortConfig> {
        return Promise.resolve({
            defaultSort: [],
            availableSorts: [],
            supported: false,
            disabled: false,
        });
    }
}
