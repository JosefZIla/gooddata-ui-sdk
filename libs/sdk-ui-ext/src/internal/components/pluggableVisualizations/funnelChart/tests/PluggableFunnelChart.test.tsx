// (C) 2019-2022 GoodData Corporation
import noop from "lodash/noop";
import { PluggableFunnelChart } from "../PluggableFunnelChart";
import * as referencePointMocks from "../../../../tests/mocks/referencePointMocks";

import { dummyBackend } from "@gooddata/sdk-backend-mockingbird";

describe("PluggableFunnelChart", () => {
    const defaultProps = {
        projectId: "PROJECTID",
        element: "body",
        configPanelElement: null as string,
        callbacks: {
            afterRender: noop,
            pushData: noop,
        },
        backend: dummyBackend(),
        visualizationProperties: {},
        renderFun: noop,
    };

    function createComponent(props = defaultProps) {
        return new PluggableFunnelChart(props);
    }

    it("should create visualization", () => {
        const visualization = createComponent();

        expect(visualization).toBeTruthy();
    });

    it("should return reference point with only one metric and one category and only valid filters", async () => {
        const funnelChart = createComponent();

        const extendedReferencePoint = await funnelChart.getExtendedReferencePoint(
            referencePointMocks.multipleMetricsAndCategoriesReferencePoint,
        );

        expect(extendedReferencePoint).toMatchSnapshot();
    });

    it("should return reference point with multiple metrics and no category", async () => {
        const funnelChart = createComponent();

        const extendedReferencePoint = await funnelChart.getExtendedReferencePoint(
            referencePointMocks.multipleMetricsNoCategoriesReferencePoint,
        );

        expect(extendedReferencePoint).toMatchSnapshot();
    });

    it("should return reference point with one metric and no category", async () => {
        const funnelChart = createComponent();
        const extendedReferencePoint = await funnelChart.getExtendedReferencePoint(
            referencePointMocks.oneMetricNoCategoriesReferencePoint,
        );

        expect(extendedReferencePoint).toMatchSnapshot();
    });

    describe("Over Time Comparison", () => {
        it("should return reference point containing uiConfig with no supported comparison types", async () => {
            const component = createComponent();

            const extendedReferencePoint = await component.getExtendedReferencePoint(
                referencePointMocks.emptyReferencePoint,
            );

            expect(extendedReferencePoint.uiConfig.supportedOverTimeComparisonTypes).toEqual([]);
        });
    });
});
