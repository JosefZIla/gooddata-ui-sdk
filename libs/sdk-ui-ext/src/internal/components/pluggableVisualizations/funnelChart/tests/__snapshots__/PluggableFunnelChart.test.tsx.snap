// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluggableFunnelChart should return reference point with multiple metrics and no category 1`] = `
Object {
  "buckets": Array [
    Object {
      "items": Array [
        Object {
          "aggregation": null,
          "attribute": "aazb6kroa3iC",
          "localIdentifier": "m1",
          "showInPercent": null,
          "showOnSecondaryAxis": null,
          "type": "metric",
        },
        Object {
          "aggregation": null,
          "attribute": "af2Ewj9Re2vK",
          "localIdentifier": "m2",
          "showInPercent": null,
          "showOnSecondaryAxis": null,
          "type": "metric",
        },
        Object {
          "aggregation": null,
          "attribute": "dt.opportunitysnapshot.snapshotdate",
          "localIdentifier": "m3",
          "showInPercent": null,
          "showOnSecondaryAxis": true,
          "type": "metric",
        },
        Object {
          "aggregation": null,
          "attribute": "acfWntEMcom0",
          "localIdentifier": "m4",
          "showInPercent": null,
          "showOnSecondaryAxis": true,
          "type": "metric",
        },
      ],
      "localIdentifier": "measures",
    },
    Object {
      "items": Array [],
      "localIdentifier": "view",
    },
  ],
  "filters": Object {
    "items": Array [],
    "localIdentifier": "filters",
  },
  "properties": Object {},
  "uiConfig": Object {
    "buckets": Object {
      "filters": Object {
        "accepts": Array [
          "attribute",
          "date",
        ],
        "allowsReordering": false,
        "enabled": true,
        "isShowInPercentEnabled": false,
        "itemsLimit": 20,
        "itemsLimitByType": Object {
          "date": 1,
        },
      },
      "measures": Object {
        "accepts": Array [
          "metric",
          "fact",
          "attribute",
        ],
        "allowsDuplicateItems": true,
        "allowsReordering": true,
        "allowsSwapping": true,
        "canAddItems": true,
        "enabled": true,
        "icon": "local:funnel/bucket-title-measures.svg",
        "isShowInPercentEnabled": false,
        "isShowInPercentVisible": false,
        "itemsLimit": 20,
        "title": "Measures",
      },
      "view": Object {
        "accepts": Array [
          "attribute",
          "date",
        ],
        "allowsReordering": false,
        "allowsSwapping": true,
        "canAddItems": false,
        "enabled": true,
        "icon": "local:funnel/bucket-title-view.svg",
        "isShowInPercentEnabled": false,
        "itemsLimit": 0,
        "itemsLimitByType": Object {
          "date": 1,
        },
        "title": "View by",
        "warningMessage": "To view by, an insight can have only one measure",
      },
    },
    "exportConfig": Object {
      "supported": true,
    },
    "openAsReport": Object {
      "supported": false,
    },
    "recommendations": Object {},
    "supportedOverTimeComparisonTypes": Array [],
  },
}
`;

exports[`PluggableFunnelChart should return reference point with one metric and no category 1`] = `
Object {
  "buckets": Array [
    Object {
      "items": Array [
        Object {
          "aggregation": null,
          "attribute": "aazb6kroa3iC",
          "localIdentifier": "m1",
          "showInPercent": null,
          "showOnSecondaryAxis": null,
          "type": "metric",
        },
      ],
      "localIdentifier": "measures",
    },
    Object {
      "items": Array [],
      "localIdentifier": "view",
    },
  ],
  "filters": Object {
    "items": Array [],
    "localIdentifier": "filters",
  },
  "properties": Object {},
  "uiConfig": Object {
    "buckets": Object {
      "filters": Object {
        "accepts": Array [
          "attribute",
          "date",
        ],
        "allowsReordering": false,
        "enabled": true,
        "isShowInPercentEnabled": false,
        "itemsLimit": 20,
        "itemsLimitByType": Object {
          "date": 1,
        },
      },
      "measures": Object {
        "accepts": Array [
          "metric",
          "fact",
          "attribute",
        ],
        "allowsDuplicateItems": true,
        "allowsReordering": true,
        "allowsSwapping": true,
        "canAddItems": true,
        "enabled": true,
        "icon": "local:funnel/bucket-title-measures.svg",
        "isShowInPercentEnabled": false,
        "isShowInPercentVisible": false,
        "itemsLimit": 20,
        "title": "Measures",
      },
      "view": Object {
        "accepts": Array [
          "attribute",
          "date",
        ],
        "allowsReordering": false,
        "allowsSwapping": true,
        "canAddItems": true,
        "enabled": true,
        "icon": "local:funnel/bucket-title-view.svg",
        "isShowInPercentEnabled": false,
        "itemsLimit": 1,
        "itemsLimitByType": Object {
          "date": 1,
        },
        "title": "View by",
      },
    },
    "exportConfig": Object {
      "supported": true,
    },
    "openAsReport": Object {
      "supported": false,
    },
    "recommendations": Object {},
    "supportedOverTimeComparisonTypes": Array [],
  },
}
`;

exports[`PluggableFunnelChart should return reference point with only one metric and one category and only valid filters 1`] = `
Object {
  "buckets": Array [
    Object {
      "items": Array [
        Object {
          "aggregation": null,
          "attribute": "aazb6kroa3iC",
          "localIdentifier": "m1",
          "showInPercent": null,
          "showOnSecondaryAxis": null,
          "type": "metric",
        },
      ],
      "localIdentifier": "measures",
    },
    Object {
      "items": Array [
        Object {
          "aggregation": null,
          "attribute": "attr.owner.department",
          "dfRef": Object {
            "uri": "a1/df",
          },
          "localIdentifier": "a1",
          "type": "attribute",
        },
      ],
      "localIdentifier": "view",
    },
  ],
  "filters": Object {
    "items": Array [
      Object {
        "aggregation": null,
        "attribute": "attr.owner.department",
        "filters": Array [
          Object {
            "attribute": "attr.owner.department",
            "displayFormRef": Object {
              "uri": "owner/attribute/df/uri",
            },
            "isInverted": false,
            "selectedElements": Array [
              Object {
                "title": "string",
                "uri": "string",
              },
            ],
            "totalElementsCount": 10,
          },
        ],
        "localIdentifier": "a1",
        "type": "attribute",
      },
    ],
    "localIdentifier": "filters",
  },
  "properties": Object {},
  "uiConfig": Object {
    "buckets": Object {
      "filters": Object {
        "accepts": Array [
          "attribute",
          "date",
        ],
        "allowsReordering": false,
        "enabled": true,
        "isShowInPercentEnabled": false,
        "itemsLimit": 20,
        "itemsLimitByType": Object {
          "date": 1,
        },
      },
      "measures": Object {
        "accepts": Array [
          "metric",
          "fact",
          "attribute",
        ],
        "allowsDuplicateItems": true,
        "allowsReordering": false,
        "allowsSwapping": true,
        "canAddItems": true,
        "enabled": true,
        "icon": "local:funnel/bucket-title-measures.svg",
        "isShowInPercentEnabled": false,
        "isShowInPercentVisible": false,
        "itemsLimit": 1,
        "title": "Measures",
      },
      "view": Object {
        "accepts": Array [
          "attribute",
          "date",
        ],
        "allowsReordering": false,
        "allowsSwapping": true,
        "canAddItems": true,
        "enabled": true,
        "icon": "local:funnel/bucket-title-view.svg",
        "isShowInPercentEnabled": false,
        "itemsLimit": 1,
        "itemsLimitByType": Object {
          "date": 1,
        },
        "title": "View by",
      },
    },
    "exportConfig": Object {
      "supported": true,
    },
    "openAsReport": Object {
      "supported": false,
    },
    "recommendations": Object {},
    "supportedOverTimeComparisonTypes": Array [],
  },
}
`;
